compose/foundation/foundation/src/skikoMain/kotlin/androidx/compose/foundation/gestures/TapGesturesDetector.skiko.kt

```diff
@@ -224,14 +227,18 @@ internal suspend fun PointerInputScope.detectRepeatingTapGestures(
     awaitEachGesture {
         val touchesCounter = ClicksCounter(viewConfiguration, clicksSlop = 50.dp.toPx())
         while (true) {
-            val down = awaitFirstDown()
-            touchesCounter.update(down)
-            val downChange = down
+            val downChange = awaitFirstDown()
+            touchesCounter.update(downChange)
             when (touchesCounter.clicks) {
                 1 -> {
                     if (onTap != null) {
-                        onTap(downChange.position)
-                        downChange.consume()
+                        awaitReleaseOrCancelled(
+                            filter = { PointerMatcher.Primary.matches(it) },
+                            consumeUntilRelease = false
+                        )?.changes?.last()?.let {
+                            onTap(it.position)
```

Comment:
The semantics of `onTap` is to be fired on `down`, not on `up`. If we need to react on `up`, we have to add `onTapRelease` (or `onClick`)

Reply:
done